src/Uno.UI/UI/Xaml/Internal/InputManager.Pointers.Managed.cs

```diff
@@ -22,72 +24,67 @@ namespace Uno.UI.Xaml.Core;
-		=> _pointerManager.SetPointerCapture(identifier);
+	partial void InitializeManagedPointers(object host)
+		=> Pointers.Init(host);
 
-	internal void ReleasePointerCapture(PointerIdentifier identifier)
-		=> _pointerManager.ReleasePointerCapture(identifier);
+	partial void InjectPointerAdded(PointerEventArgs args)
+		=> Pointers.InjectPointerAdded(args);
 
-	private PointerManager _pointerManager = null!;
+	partial void InjectPointerUpdated(PointerEventArgs args)
+		=> Pointers.InjectPointerUpdated(args);
 
-	partial void InitializeManagedPointers()
-	{
-		_pointerManager = new PointerManager(this);
-	}
+	partial void InjectPointerRemoved(PointerEventArgs args)
+		=> Pointers.InjectPointerRemoved(args);
 
-	private class PointerManager
+	internal class PointerManager
 	{
-		private static IPointerExtension? _pointerExtension;
-
 		// TODO: Use pointer ID for the predicates
 		private static readonly StalePredicate _isOver = new(e => e.IsPointerOver, "IsPointerOver");
 
 		private readonly Dictionary<Pointer, UIElement> _pressedElements = new();
 
 		private readonly InputManager _inputManager;
+		private IUnoCorePointerInputSource? _source;
 
 		public PointerManager(InputManager inputManager)
 		{
-			if (_pointerExtension is null)
-			{
-				ApiExtensibility.CreateInstance(typeof(PointerManager), out _pointerExtension); // TODO: Add IPointerExtension implementation to all Skia targets and create instance per XamlRoot https://github.com/unoplatform/uno/issues/8978
-			}
 			_inputManager = inputManager;
+		}
 
-			if (_inputManager._contentRoot.Type == ContentRootType.CoreWindow)
+		/// <summary>
+		/// Initialize the InputManager.
+		/// This has to be invoked only once the host of the owning ContentRoot has been set.
+		/// </summary>
+		public void Init(object host)
+		{
+			if (!ApiExtensibility.CreateInstance(host, out _source))
 			{
-				Windows.UI.Xaml.Window.Current.CoreWindow.PointerMoved += (c, e) => OnPointerMoved(e);
-				Windows.UI.Xaml.Window.Current.CoreWindow.PointerEntered += (c, e) => OnPointerEntered(e);
-				Windows.UI.Xaml.Window.Current.CoreWindow.PointerExited += (c, e) => OnPointerExited(e);
-				Windows.UI.Xaml.Window.Current.CoreWindow.PointerPressed += (c, e) => OnPointerPressed(e);
-				Windows.UI.Xaml.Window.Current.CoreWindow.PointerReleased += (c, e) => OnPointerReleased(e);
-				Windows.UI.Xaml.Window.Current.CoreWindow.PointerWheelChanged += (c, e) => OnPointerWheelChanged(e);
-				Windows.UI.Xaml.Window.Current.CoreWindow.PointerCancelled += (c, e) => OnPointerCancelled(e);
+				throw new InvalidOperationException("Failed to initialize the PointerManager: cannot resolve the IUnoCorePointerInputSource.");
 			}
+
+			// Currently there is no need to filter on ContentRootType ('_inputManager._contentRoot.Type is ContentRootType.CoreWindow')
+			// As soon as we have a CoreWindow we configure it.
+			// This might needed later once we support multi windowing
```

Comment:
```suggestion
			// Currently there is no need to filter on ContentRootType ('_inputManager._contentRoot.Type is ContentRootType.CoreWindow')
			// As soon as we have a CoreWindow we configure it.
			// This might be needed later once we support multi-windowing
```